storage/allocator.go

```diff
@@ -81,6 +93,120 @@ func getUsedNodes(existing []proto.Replica) map[proto.NodeID]struct{} {
 	return usedNodes
 }
 
+// findDeadReplicas returns a list of nay replicas in the given RangeDescriptor
+// which are located on a dead store.
+func (a *allocator) findDeadReplicas(desc *proto.RangeDescriptor) []proto.Replica {
+	var deadReplicas []proto.Replica
+	// TODO(mrtracy): Pull in Bram's storePool changes to support this.
+	/*
+		for _, repl := range desc.Replicas {
+			if a.storePool.getStoreDetail(repl.StoreID).dead {
+				deadReplicas = append(deadReplicas, repl)
+			}
+		}
+		return deadReplicas
+	*/
+	return deadReplicas
+}
+
+// rangeAvailability computes some availability statistics about a range that
+// has dead replicas.
+func (a *allocator) computeRangeAvailability(desc *proto.RangeDescriptor, dead []proto.Replica) (bool, int) {
+	quorum := (len(desc.Replicas) / 2) + 1
```

Comment:
The target quorum isn't the important thing here; we are considering whether the current raft group can make progress (if a quorum of the current replicas isn't available, then no replicas can be added or removed).